RoddyCore/src/de/dkfz/roddy/execution/io/fs/BashCommandSet.groovy

```diff
@@ -122,20 +127,47 @@ class BashCommandSet extends ShellCommandSet {
             return getCheckAndCreateDirectoryCommand(file, onCreateFileGroup, onCreateAccessRights) + TRUE_OR_FALSE;
     }
 
-    @Override
-    String getCheckAndCreateDirectoryCommand(File f, String onCreateAccessRights, String onCreateFileGroup) {
-        String path = f.absolutePath
+    /** If onCreateAccessRights and onCreateFileGroup are both != null, change to the group and create the path
+     *  using the group and access rights. However, if the path already exists (file or whatever), do nothing
+     *  (`umask $mask && mkdir -p $path` does nothing on an existing directory) or fail, if the path is occupied
+     *  by a file rather than a directory.
+     *
+     *  If one of onCreateAccessRights and onCreateFileGroup is null, do nothing if the file/path exists. If the
+     *  file does not exist, create a directory. If the directory creation fails, ignore the error.
+     *
+     * @param file
+     * @param onCreateAccessRights
+     * @param onCreateFileGroup
+     * @return
+     */
+    @Override
+    String getCheckAndCreateDirectoryCommand(File file, String onCreateAccessRights, String onCreateFileGroup) {
+        String path = file.absolutePath
         String checkExistence = "[[ -e ${path} ]]"
         if (onCreateAccessRights && onCreateFileGroup)
             return "sg ${onCreateFileGroup} -c \"${checkExistence} || umask ${onCreateAccessRights} && mkdir -p ${path}\""
         else
+            // I do not understand, why this command ignores execution errors (via `|| echo ''`).
             return "${checkExistence} || install -d \"${path}\" || echo ''"
     }
 
-    @Override
-    String getCheckChangeOfPermissionsPossibilityCommand(File f, String group) {
-        File testFile = new File(f, ".roddyPermissionsTestFile")
-        return "(touch ${testFile}; chmod u+rw ${testFile} &> /dev/null && chgrp ${group} ${testFile} &> /dev/null) $TRUE_OR_FALSE; rm ${testFile} 2>/dev/null; echo ''"
+    /** Check whether it is possible to change permissions at the target-location. Specifically, the path `file` will
+     *  be appended by ".roddyPermissionTestFile" and it is checked, whether this can get changes of the permissions.
+     *  Currently, it is just checked whether chmod is working on that file and -- if group is not null -- whether
+     *  the group ownership can be changed to the target group.
+     *
+     * @param file    path to check. Not exactly this path is changed, but a file besides that path.
+     * @param group   target-group to check; if this is null, then only chmod is tested, otherwise also chgrp.
+     * @return
+     */
+    @Override
+    String getCheckChangeOfPermissionsPossibilityCommand(File file, String group) {
+        File testFile = new File(file, ".roddyPermissionsTestFile")
+        if (null == group) {
```

Comment:
is Yoda conditions default in Roddy? Seen this at various positions.

Reply:
I try to use these "Yoda conditions" (thanks for the term) to not by chance write `group = null`. Note that `group == null` evaluates to `true` if group is null, but `group = null` evaluates to `false`. So, in case of a typo with `=` rather than `==`, not only `group` would be assigned, but also the other branch would be taken. Just a safety measure.

Reply:
Actually, I never checked, but apparently Groovy does not accept assignments in conditions. So it is not strictly unnecessary. However, the "added cognitive load" counter-argument on the Wiki page is ridiculously irrelevant in the context of the overall complexity of the code. So I leave it as it is but avoid it in the future. Thanks for pointing out.